assets/app/scripts/filters/date.js

```diff
@@ -7,6 +7,47 @@ angular.module('openshiftConsole')
       return moment(timestamp).fromNow();
     };
   })
+  .filter('duration', function() {
+    return function(timestampLhs, timestampRhs) {
+      if (!timestampLhs) {
+        return timestampLhs;
+      }
+      timestampRhs = timestampRhs || Date();
+
+      var ms = moment(timestampRhs).diff(timestampLhs);
+      var duration = moment.duration(ms);
+      // the out of the box humanize in moment.js rounds to the nearest time unit
+      // but we need more details
+      var humanizedDuration = "";
+      if (duration.years() > 0) {
+        humanizedDuration += duration.years() + " years";
+      }
+      if (duration.months() > 0) {
+        if (humanizedDuration.length > 0) { humanizedDuration += ", "; }
+        humanizedDuration += duration.months() + " months";
+      }
+      if (duration.days() > 0) {
+        if (humanizedDuration.length > 0) { humanizedDuration += ", "; }
+        humanizedDuration += duration.days() + " days";
+      }
+      if (duration.hours() > 0) {
+        if (humanizedDuration.length > 0) { humanizedDuration += ", "; }
+        humanizedDuration += duration.hours() + " hrs";
```

Comment:
just hours? or do you not like minutes and seconds being shortened either?  just seemed to take up a lot of space saying '"5 minutes, 6 seconds"